uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -297,8 +297,8 @@ define([
      * Sets the "change_balance" attribute to the leave request
      * if a force balance change recalculation is needed
      */
-    function decideIfBalanceChangeNeedsAForceRecalculation () {
-      if (!vm.isRole('staff')) {
+    function decideIfBalanceChangeNeedsRecalculationOnBackend () {
+      if (isBalanceChangeRecalculationNeeded() && !vm.isRole('staff')) {
```

Comment:
I want to understand how this `change_balance` property is fetched. Because it will depend on the status which you set before saving, so do we get that while calling `isValid()`?